WordPress/Classes/Models/AbstractPost.m

```diff
@@ -572,6 +572,11 @@ - (BOOL)hasLocalChanges
         return YES;
     }
 
+    if ( ((self.featuredImage != nil) && ![self.featuredImage.objectID isEqual: original.featuredImage.objectID]) ||
```

Comment:
What do you think about adding unit tests to ensure this logic? Under `PostTests`.

Something like:

```
    /// When removing the featured image hasLocalChanges returns true
    func testLocalChangesWhenfeaturedImageIsRemoved() {
        let post = newTestPost()
        post.featuredImage = Media.makeMedia(in: context)
        let revision = post.createRevision()
        revision.featuredImage = nil

        XCTAssertTrue(revision.hasLocalChanges())
    }
```

And besides that test we can:

1. Test when the post originally didn't had any featured image and the user adds one
2. The user sets the same featured image (based on objectID) and hasLocalChanges returns false

Reply:
Good call, I will add the tests to it!